torch_xla/csrc/ops/ops_lower_fn.cpp

```diff
@@ -344,11 +344,19 @@ torch_xla::XlaOpVector ClampMinTensor::Lower(LoweringContext* loctx) const {
 
 torch_xla::XlaOpVector Cos::Lower(LoweringContext* loctx) const {
   xla::XlaOp xla_input = loctx->GetOutputOp(operand(0));
+  if (xla::primitive_util::IsIntegralType(XlaHelpers::TypeOfXlaOp(xla_input))) {
+    xla::PrimitiveType input_type = XlaHelpers::TypeOfXlaOp(xla_input);
+    xla_input = ConvertTo(xla_input, input_type, xla::PrimitiveType::F32);
```

Comment:
Do you know why we want to convert the input type to F32 when it is the integral type?

Reply:
It was to address a feature gap in torch and torch_xla. In torch, `aten::cos` allows integer types as inputs. However, torch_xla didn't. This PR fixes that. More details at https://github.com/pytorch/xla/pull/6110.